include/openssl/core_mkdigest.h

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_CORE_MKDIGEST_H
+# define OSSL_CORE_MKDIGEST_H
```

Comment:
Hmmm.... I don't think this header file should be part of the public API.

Reply:
It looks like it could be useful to providers to define digests.  Is that enough reason to make it public?  How many different digests are other providers like to have?  Third party FIPS seems like the worst case: SHA1, SHA2, SHA3, SHAKE in all flavours (about a dozen in total).  They'd also need the _exact_ same function wrappings as we do (which doesn't seem all that unlikely)

I'm thinking private is okay.  We can make it public later if enough provider writers want it.

Reply:
The keys bit of it has been moved into core_names

Reply:
I have kept the filename the same and moved it into provider/include/internal for now.